portfolio/src/main/webapp/quiz.js

```diff
@@ -2,40 +2,125 @@
  * Question data type.
  */
 class Question {
-  constructor(question, answers, correct) {
+  /**
+   * @param {string} question The question that shall be posed
+   * @param {Array<string>} answers The potential responses to the question
+   * @param {number} correctAnswerIndex The index in answers that leads to the correct response
+   */
+  constructor(question, answers, correctAnswerIndex) {
+    /**
+     * @private @const {string}
+     */
     this.question_ = question;
+    /**
+     * @private @const {Array<string>}
+     */
     this.answers_ = answers;
+    /**
+     * @private @const {number}
+     */
     this.correctAnswerIndex_ = correctAnswerIndex;
   }
 
+  /**
+   * @return {string}
+   */
   getQuestion() {
-    return question_;
+    return this.question_;
   }
 
+  /**
+   * @return {Array<string>}
+   */
   getAnswers() {
-    return answers_;
+    return this.answers_;
   }
 
+  /**
+   * @return {number}
+   */
   getCorrectAnswerIndex() {
-    return correctAnswerIndex_;
+    return this.correctAnswerIndex_;
   }
 }
 
+const questionsArray = [];
+questionsArray.push(
+    new Question(question='What\'s my GitHub username?',
```

Comment:
nit - feel free to use a [template string](https://google.github.io/styleguide/jsguide.html#jsdoc-nullability) to avoid escaping these single quotes.

Reply:
Ok, will do.